### ISSUE TITLE
Nomad metric nomad.client.allocs.memory.rss in docker with cgroup v2 - unavailable

### ISSUE THREAD

**Issue by Alexsandr-Random** [hashicorp/nomad#19185]
### Nomad version
Any

### Operating system and Environment details
NAME="Ubuntu"
VERSION="22.04 LTS"

### Issue
When we using cgroups v2 nomad agent stops sending some metrics for Prometheus. Most important for us is nomad.client.allocs.memory.rss
When cgroups set to v1 - everything scraping correctly.
I do researches on 3 different independent nomad clusters so i could say this looks like bug.

### Reproduction steps
Start some docker job on latest versions on ubuntu when cgroups v2 enabled by default and enable telemetry stanza on nomad client to send all metrics to Prometheus. And try to check if nomad.client.allocs.memory.rss scraping correctly  
Then start the same job with cgroups v1 and you will get again nomad.client.allocs.memory.rss 

#### Expected Result
nomad.client.allocs.memory.rss - scraping from any ubuntu version and from any cgroups version.

#### Actual Result
nomad.client.allocs.memory.rss - scraping only from cgroups v1.

**Comment by jrasell** [comment]
Hi @Alexsandr-Random and thanks for raising this issue.

The memory implementation of [cgroups v2](https://www.kernel.org/doc/html/latest/admin-guide/cgroup-v2.html#memory-interface-files) does not expose RSS and is therefore not supported by the Docker driver. The difference between the memory statistics available within the Docker driver can be [seen here](https://github.com/hashicorp/nomad/blob/591394fb62f828917e57055c7423162e4ede524f/drivers/docker/util/stats_posix.go#L18-L19).

It seems there might an equivalent when interrogating the cgroup v2 memory files, however, I am unsure if it would be possible to plumb this through to the Docker driver. This is because it does not have any direct understanding of the isolation which is left to Docker itself. I'll keep this issue open for future readers and mark it that it requires further investigation.

**Comment by Alexsandr-Random** [comment]
@jrasell Thanks for quick response! 
So for now the only solution to get metrics like memory rss on latest distros is manually downgrade cgroups v2 --> v1?

**Comment by tgross** [comment]
See also https://github.com/hashicorp/nomad/issues/19604

**Comment by Alexsandr-Random** [comment]
Yes, I can confirm what is written in https://github.com/hashicorp/nomad/issues/19604.

Nomad indeed displays accurate metrics (nomad_client_allocs_memory_usage) only when using cgroups v2 or higher.

If cgroups = v1, the correct memory percentage metrics would be the ratio of 

```
nomad_client_allocs_memory_rss / nomad_client_allocs_memory_allocated.
```

Otherwise, you may observe behavior where memory is either immediately ~= 100% or it may appear to be a memory leak (on graphs), although that is not the case.

When using the expression below you might encounter inaccurate representations.
```
nomad_client_allocs_memory_usage / nomad_client_allocs_memory_allocated,
```

**Comment by tgross** [comment]
I've got https://github.com/hashicorp/nomad/pull/25751 open in draft which should resolve this issue. After a bit of digging I was able to determine the equivalent values in the cgroups v2 memory stats and we'll be able to expose that for the `docker`, `exec`, and `java` drivers.

Once that's reviewed and merged we'll release it in the next regular release of Nomad (likely 1.10.1) as well as backports to Nomad Enterprise.
